euphoria-core/src/main/java/cz/seznam/euphoria/core/client/functional/UnaryFunctionEnv.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2016-2017 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.core.client.functional;
+
+import cz.seznam.euphoria.core.client.io.Context;
+
+import java.io.Serializable;
+
+/**
+ * Function of single argument with access to Euphoria environment via context.
+ *
+ * @param <IN> the type of the element processed
+ * @param <OUT> the type of the result applying element to the function
+ */
+@FunctionalInterface
+public interface UnaryFunctionEnv<IN, OUT> extends Serializable {
```

Comment:
I'd suggest naming this interface something like `RichUnaryFunction`, because in conjunction with @136, we could add additional methods (with default implementation) here.